master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -677,6 +677,7 @@ func (p k8sJobResource) Start(
 		spec:         spec,
 		slots:        p.slots,
 		rank:         rri.AgentRank,
+		resourcePool: p.req.ResourcePool,
```

Comment:
I think I found a bug here -- we reference this field when we start a job in `master/internal/rm/kubernetesrm/jobs.go`, but we never set it on the resource pool level. 
In fact, we only set this field when we attempt to restore resources/reattach a job.